src/main/java/fi/okm/jod/yksilo/config/SessionConfig.java

```diff
@@ -58,7 +62,23 @@ public RedisSerializer<Object> springSessionDefaultRedisSerializer() {
   }
 
   @Bean
-  @Profile("cloud")
+  public SpringSessionBackedSessionRegistry<S> sessionRegistry(
+      FindByIndexNameSessionRepository<S> sessionRepository) {
+    return new SpringSessionBackedSessionRegistry<>(sessionRepository);
+  }
+
+  @Bean
+  @Order(2)
+  public SecurityFilterChain sessionFilterChain(
+      HttpSecurity http, SpringSessionBackedSessionRegistry<S> sessionRegistry) throws Exception {
+    return http.sessionManagement(
+            (sessionManagement) ->
+                sessionManagement.maximumSessions(1).sessionRegistry(sessionRegistry))
+        .build();
+  }
```

Comment:
I'm not sure if using generics here is justified or desired -- we anyway want to use the concrete Redis session repository, and should there be more that one qualifying bean, this would fail.

```suggestion
  public SessionRegistry sessionRegistry(RedisIndexedSessionRepository sessionRepository) {
    return new SpringSessionBackedSessionRegistry<>(sessionRepository);
  }

  @Bean
  @Order(2)
  public SecurityFilterChain sessionFilterChain(HttpSecurity http) throws Exception {
    return http.sessionManagement((sessionManagement) -> sessionManagement.maximumSessions(1))
        .build();
  }
```

Reply:
Should be fine. See https://docs.spring.io/spring-session/reference/3.0/spring-security.html#spring-security-concurrent-sessions.